### ISSUE TITLE
Texture decoding changes

### ISSUE THREAD

**Issue by robert-nix** [HearthSim/UnityPack#11]
patch for use with my dxtc branch of Pillow; need to make a PR when #2068 is merged

``` diff
diff --git a/unitypack/engine/texture.py b/unitypack/engine/texture.py
index 8f87fc7..1434854 100644
--- a/unitypack/engine/texture.py
+++ b/unitypack/engine/texture.py
@@ -1,6 +1,5 @@
 import struct
 from enum import IntEnum
-from io import BytesIO
 from .object import Object, field


@@ -117,6 +116,7 @@ class Texture2D(Texture):
    texture_dimension = field("m_TextureDimension")
    mipmap = field("m_MipMap")
    complete_image_size = field("m_CompleteImageSize")
+   stream_data = field("m_StreamData")

    def __repr__(self):
        return "<%s %s (%s %ix%i)>" % (
@@ -124,17 +124,11 @@ class Texture2D(Texture):
        )

    @property
-   def decoded_data(self):
-       from PIL.DdsImagePlugin import _dxt1, _dxt5
+   def image_data(self):
+       if self.stream_data and self.stream_data['path']:
+           return self.asset.environment.get_stream(self.stream_data['path'], self.stream_data['offset'], self.stream_data['size'])

-       if self.format == TextureFormat.DXT1:
-           codec = _dxt1
-       elif self.format == TextureFormat.DXT5:
-           codec = _dxt5
-       else:
-           return self.data
-
-       return codec(BytesIO(self.data), self.width, self.height)
+       return self.data

    @property
    def image(self):
@@ -146,9 +140,17 @@ class Texture2D(Texture):
        size = (self.width, self.height)
        raw_mode = self.format.pixel_format
        mode = "RGB" if raw_mode == "RGB" else "RGBA"
-       data = bytes(self.decoded_data)
+       codec = "raw"
+       args = (raw_mode)
+       if self.format == TextureFormat.DXT1:
+           codec = "bcn"
+           args = (1)
+       elif self.format == TextureFormat.DXT5:
+           codec = "bcn"
+           args = (3)
+       data = bytes(self.image_data)

        if not data and size == (0, 0):
            return None

-       return Image.frombytes(mode, size, data, "raw", raw_mode)
+       return Image.frombytes(mode, size, data, codec, args)
```

**Comment by jleclanche** [comment]
What's streamData? Is it new?

**Comment by robert-nix** [comment]
Yep.  It is the exact same bytes as `image data` but exists in a streamed resource instead of alongside the object data.

**Comment by jleclanche** [comment]
So it won't work for older versions of unity? This has to work with at the very least the first hearthstone unity3d files.

**Comment by robert-nix** [comment]
It does work, but only because it reads past the end of old-format object data (ObjectInfo.size isn't used to limit the reading)

**Comment by jleclanche** [comment]
Reworked and landed in ad149d4f1ae65dd57d045714ff97c27334c40eb1
